### PR TITLE
(fix) Modified NewsML 1.2 and NewsML G2 Parser to handle Media Publishing

### PR DESCRIPTION
**Pull request by mdhaman**
- Modified Parser handle Category, Place/Locator, Media Type.
- Fixed other serialization issue.

This PR requires https://github.com/superdesk/superdesk-core/pull/22

For, We need to define Catalog for NewsML1.2 and NewsML G2 formatter and that will be separate PR.

### CODE REVIEW

File: server/apps/publish/formatters/newsml_1_2_formatter.py [hunk 1]
```diff
@@ -26,13 +34,30 @@ class NewsML12Formatter(Formatter):
...
+        :return [(int, str)]: return a List of tuples. A tuple consist of
+            publish sequence number and formatted article string.
+        :raises FormatterError: if the formatter fails to format an article
+        """
         try:
             pub_seq_num = superdesk.get_resource_service('subscribers').generate_sequence_number(subscriber)
 
             newsml = etree.Element("NewsML")
-            SubElement(newsml, "Catalog", {'Href': 'http://www.aap.com.au/xml-res/aap-master-catalog.xml'})
+            SubElement(newsml, "Catalog", {'Href': 'http://www.iptc.org/std/catalog/catalog.IptcMasterCatalog.xml'})
```
Comment (plamut) [inline server/apps/publish/formatters/newsml_1_2_formatter.py]: Does `'Href'` need to be capitalized? Or it doesn't really matter at all?

**Comment by sivakuna-aap** [comment]
:+1:

**Comment by amagdas** [comment]
:+1: The core PR is merged, we should update requirements.

**Comment by marwoodandrew** [comment]
:+1:

File: server/apps/publish/formatters/newsml_1_2_formatter.py (hunk 1 above)
Comment (mdhaman) [inline server/apps/publish/formatters/newsml_1_2_formatter.py]: yes. it needs to capitalized as per newsml1.2 schema

**Pull request merged**